proposals/container-build-plan-spec.md

```diff
@@ -0,0 +1,176 @@
+# Container Build Plan Specification
+
+Specification for building a container image.
+
+Although looking similar, the structure and semantics of similary named properties are different from the Docker/[OCI Image Configuration](https://github.com/opencontainers/image-spec/blob/master/config.md).
+
+## Example
+
+```
+{
+  "baseImage": "gcr.io/distroless/java@sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9",
+  "architecture": "amd64",
+  "os": "linux",
+  "format": "Docker",
+  "created": "2011-12-03T22:42:05Z",
+
+  "config": {
+    "env": {
+      "KEY": "value",
+      "PATH": "/usr/sbin:/usr/bin:/sbin:/bin",
+      "HOME": "/home/guest"
+    },
+    "labels": {
+      "com.example.department.some-label-key": "avocado explosion",
+      "com.example.system.xml": "<message>delivered</message>"
+    },
+    "volumes": ["/mnt/shared", "/tmp"]
+    "exposedPorts": ["8080", "53/udp", "80/tcp"]
+    "user": ":12345",
+    "workingDir": "/
+    "entrypoint": ["/bin/bash", "-c"],
+    "cmd": [
+      "-x",
+      "set -o errexit ; echo \"\$0=$0 \$1=$1\" ; echo \"\$PATH=${PATH}\" ; exit 0",
+      "my-shell-name",
+      "first shell arg"
+    ],
+  },
+
+  "layers": [
+    {
+      "entries": [
+        {
```

Comment:
could we have nested file-permission/owner configs? like in go/cbsp?

Reply:
As discussed, we will consider such extension in the future.

Reply:
Again, at a different layer.